tests/integration_tests/functional/test_mmds.py

```diff
@@ -342,6 +361,51 @@ def test_larger_than_mss_payloads(test_microvm_with_ssh, network_config):
     _assert_out(stdout, stderr, lower_than_mss)
 
 
+@pytest.mark.parametrize(
```

Comment:
I would also add a negative unit test like for example if the metadata file is missing or bad permissions.